pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/SegmentMetadataImpl.java

```diff
@@ -403,6 +408,10 @@ public Map<String, ColumnMetadata> getColumnMetadataMap() {
     return _columnMetadataMap;
   }
 
+  public boolean isUpsertSnapshotEnabled() {
```

Comment:
We should not rely on segment metadata to determine whether snapshot is enabled. The upsert config might change anytime, and we want to use/drop snapshot accordingly.

Reply:
Gotcha, thanks for the suggestion! let me address this.